analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/SecureRandomSeedsShouldNotBePredictable.cs

```diff
@@ -94,16 +115,16 @@ private static ProgramState ProcessSeedingMethods(ProgramState state, IInvocatio
             && invocation.Instance is { } instance
             // If it is already unpredictable, do nothing.
             // Seeding methods do not overwrite the state, but _mix_ it with the new value.
-            && state[instance]?.HasConstraint(CryptographicSeedConstraint.Unpredictable) is false
-            && invocation.ArgumentValue("seed") is { } argumentValue
+            && state[instance]?.HasConstraint(CryptographicSeedConstraint.Predictable) is true
+            && SeedValue() is { } seedValue
```

Comment:
There are two things happening here:

### I swapped the conditions, and now I am entering this if ONLY if the instance is already predictable.

`state[instance]?.HasConstraint(CryptographicSeedConstraint.PREDICTABLE) is TRUE`

There are 3 cases:
- Predictable
- Unpredictable
- No constraint

No constraint and Unpredictable should behave the same, as unpredictable. This is the case, for example, when we get the `SecureRandom` or the `RandomGenerator` as an argument to the current method.
In this case, we have no constraint and we need to treat it as unknown, therefore Unpredictable.

So I made the logic positive, to enter the body ONLY if we 100% know this is predictable, and the `AddSeedMaterial` or `SetSeed` might change that.

### SeedValue became more complex

Check `SeedValue`

There are three types we might care about here:

- Vmprc generator
- Digest generator
- IGenerator interface

Apparently, they decided to change the _name_ of the argument for `Digest`, so now it is not called `seed`, as it is called in the interface it implements.
(This makes me slightly upset.)